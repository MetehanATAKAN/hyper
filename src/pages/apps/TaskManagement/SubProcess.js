import React, { useState } from 'react';
import { Box, Tab } from '@mui/material';
import { TabContext, TabList, TabPanel } from '@mui/lab';
import '../../../assets/scss/custom/taskmanagement/subProcess.scss';
import Name from './subProcess/Name';
import Approve from './subProcess/Approve';
import Gant from './subProcess/Gant';
import Estimated from './subProcess/Estimated';
import CheckList from './subProcess/CheckList';
import { useTranslation } from 'react-i18next';

const SubProcess = (props) => {
    const { setOnModal, subProcess, activityTypes, setSubProcess, parentProcess, setParentProcess,onModal } = props;
    const { t } = useTranslation();
    const [tabValue, setTabValue] = useState('1');
    const tabChange = (event, newValue) => {
        setTabValue(newValue);
    };

    const tabStyle = {
        color: '#5d636a',
        textTransform: 'none',
        fontWeight: '600',
    };
    return (
        <>
         {
            onModal &&
            <Box sx={{ width: '100%', typography: 'body1' }}>
            <TabContext value={tabValue}>
                <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
                    <TabList variant="fullWidth" id="process-tabs" textColor="primary" onChange={tabChange}>
                        <Tab style={tabStyle} label={t('Name')} value="1" />
                        <Tab style={tabStyle} label={t('Approve')} value="2" />
                        <Tab style={tabStyle} label={t('Gant')} value="3" />
                        <Tab style={tabStyle} label={t('Estimated')} value="4" />
                        <Tab style={tabStyle} label={t('Check List')} value="5" />
                    </TabList>
                </Box>
                {/* Name */}
                <TabPanel value="1" id="process-tabs__content">
                    <Name
                        setOnModal={setOnModal}
                        subProcess={subProcess}
                        activityTypes={activityTypes}
                        setSubProcess={setSubProcess}
                        parentProcess={parentProcess}
                        setParentProcess={setParentProcess}
                        setTabValue={setTabValue}
                    />
                </TabPanel>
                {/* Approve */}
                <TabPanel value="2" id="process-tabs__content">
                    <Approve setOnModal={setOnModal} setTabValue={setTabValue} />
                </TabPanel>
                {/* Gantt */}
                <TabPanel value="3" id="process-tabs__content">
                    <Gant setOnModal={setOnModal} subProcess={subProcess} setTabValue={setTabValue} />
                </TabPanel>
                {/* Estimated */}
                <TabPanel value="4" id="process-tabs__content">
                    <Estimated setOnModal={setOnModal} setTabValue={setTabValue} />
                </TabPanel>
                {/* Check List */}
                <TabPanel value="5" id="process-tabs__content">
                    <CheckList setOnModal={setOnModal} />
                </TabPanel>
            </TabContext>
        </Box>
         }
        </>
    );
};

export default React.memo(SubProcess);
