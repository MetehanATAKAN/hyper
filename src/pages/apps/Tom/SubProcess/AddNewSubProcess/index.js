import React from 'react'
import NewSubProcessHeader from './NewSubProcessHeader'
import NewSubProcessContent from './NewSubProcessContent'
import { Col, Row } from 'antd'
import LeftBarMenu from './LeftBarMenu'
import { useSelector } from 'react-redux'
import ActionList from './ActionList'
import ActionListHeader from './ActionList/ActionListHeader'

const NewSubProcess = () => {
  
  const leftBarId = useSelector(state => state.SubProcess.leftBarKey);
  console.log(leftBarId);
  return (
    <>
      {
        leftBarId === 0
          ? <NewSubProcessHeader />
          : leftBarId === 3
            ? <ActionListHeader />
            : <NewSubProcessHeader />
      }
      <Row className='new-sub-process' gutter={16}>
        <Col span={6}>
          <LeftBarMenu />
        </Col>
        <Col className='new-sub-process-right-content' span={18}>
          {
            leftBarId === 0
              ? <NewSubProcessContent />
              : leftBarId === 3
                ? <ActionList />
                : <NewSubProcessContent />
          }
        </Col>
      </Row>
    </>
  )
}

export default NewSubProcess