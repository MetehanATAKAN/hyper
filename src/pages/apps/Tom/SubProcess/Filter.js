import React, { useEffect, useState } from 'react'
import Filters from '../../../../components/Filter';
import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
import { FetchApiGet, FetchApiPost } from '../../../../utils/http.helper';

const Filter = ({
    isApply,
    setIsApply,
    setTableLoading,
    setTableData
}) => {

    const history = useHistory();

    /**activity type */
   const [activityType, setActivityType] = useState([]);
   const [selectActivityType, setSelectActivityType] = useState([]);

    /**process */
    const [process, setProcess] = useState([]);
    const [selectProcess, setSelectProcess] = useState([]);

    /**sub process */
    const [subProcess, setSubProcess] = useState([]);
    const [selectSubProcess, setSelectSubProcess] = useState([]);

    /**mix type */
    const [mixType, setMixType] = useState([
        {
            value:1,
            label:'Product'
        },
        {
            value:2,
            label:'Position'
        },
        {
            value:3,
            label:'Employee'
        },
        {
            value:4,
            label:'Process'
        },
    ]);
    const [selectMixType, setSelectMixType] = useState([
        {
            value:1,
            label:'Product'
        },
        {
            value:2,
            label:'Position'
        },
        {
            value:3,
            label:'Employee'
        },
        {
            value:4,
            label:'Process'
        },
    ]);

    /**status */
    const [status, setStatus] = useState([
        {
            value:1,
            label:'Redact'
        }
    ]);
    const [selectStatus, setSelectStatus] = useState([
        {
            value:1,
            label:'Redact'
        }
    ]);

    const filter = [
        {
            key: 'activityType',
            label: 'activity type',
            options: activityType,
            state: selectActivityType,
            setState: setSelectActivityType,
            type: 'multiselect',
            maxSelect:false,
            maxSelectItem:false
        },
        {
            key: 'process',
            label: 'process',
            options: process,
            state: selectProcess,
            setState: setSelectProcess,
            type: 'multiselect',
            maxSelect:false,
            maxSelectItem:false
        },
        {
            key: 'mixType',
            label: 'mix type',
            options: mixType,
            state: selectMixType,
            setState: setSelectMixType,
            type: 'multiselect',
            maxSelect:false,
            maxSelectItem:false
        },
        {
            key: 'status',
            label: 'status',
            options: status,
            state: selectStatus,
            setState: setSelectStatus,
            type: 'multiselect',
            maxSelect:false,
            maxSelectItem:false
        },
    ]

    const apply = () => {
        setIsApply(false);
        setTableLoading(true);
        if (selectActivityType.length > 0 &&
            selectProcess.length > 0 &&
            selectStatus.length > 0 &&
            selectMixType.length > 0
        ) {
            const body = {
                activityTypeIds: selectActivityType?.map(data => data.value),
                processIds: selectProcess?.map(data => data.value),
                statusIds: selectStatus?.map(data => data.value),
                mixTypes: selectMixType?.map(data => data.label)
            }
            FetchApiPost('services/TaskManagement/SubProcess/GetNewSubProcessForApply', 'POST', body)
                .then((res) =>
                    (async () => {
                        try {
                            if (res.status === 200) {
                                setTableLoading(false);
                                res.json().then(el => {
                                    setTableData(el.data.map(data => (
                                        {
                                            id:data.id,
                                            subProcessId:data.subId,
                                            subProcessName:data.name,
                                            day:data.approverExpectedDay,
                                            activityType:data.activityType.title,
                                            mixType:data.subProcessMixType,
                                            process:'-', //processes
                                            status:data.approveStatus === 1 && 'Redact'
                                        }
                                    )))
                                })

                            }
                            else if (res.status === 500 || res.status === 499) {
                                setTableLoading(false);
                                history.push('/error-500');
                            }

                        } catch (error) {
                            console.log('error', error);
                        }
                    })()
                )
        }
    }

    const clearFilter = () => {
        console.log('clear filter da');
    }
    /**activity type */
    useEffect(() => {

        FetchApiGet('services/TaskManagement/ActivityType/GetAllActivityTypes', 'GET')
            .then((res) =>
                (async () => {
                    try {
                        if (res.status === 200) {
                            ;
                            res.json().then(el => {
                                setActivityType(el.data?.map(data => (
                                    {
                                        value: data.id,
                                        label: data.title
                                    }
                                )))
                                setSelectActivityType(el.data?.map(data => (
                                    {
                                        value: data.id,
                                        label: data.title
                                    }
                                )))
                            })

                        }
                        else if (res.status === 500 || res.status === 499) {

                            history.push('/error-500');
                        }

                    } catch (error) {
                        console.log('error', error);
                    }
                })()
            )

    }, [history])

    /**process */
    useEffect(() => {

        FetchApiGet('services/TaskManagement/ParentProcess/GetAllParentProcess', 'GET')
            .then((res) =>
                (async () => {
                    try {
                        if (res.status === 200) {
                            ;
                            res.json().then(el => {
                                setProcess(el.data?.map(data => (
                                    {
                                        value: data.id,
                                        label: data.title
                                    }
                                )))
                                setSelectProcess(el.data?.map(data => (
                                    {
                                        value: data.id,
                                        label: data.title
                                    }
                                )))
                            })

                        }
                        else if (res.status === 500 || res.status === 499) {

                            history.push('/error-500');
                        }

                    } catch (error) {
                        console.log('error', error);
                    }
                })()
            )

    }, [history])

    /**apply */
    useEffect(() => {
      isApply &&(
        apply()
      )
    }, [isApply])
    
  return (
    <div>
        <Filters
        filterComponentsData={filter}
        />
    </div>
  )
}

export default Filter