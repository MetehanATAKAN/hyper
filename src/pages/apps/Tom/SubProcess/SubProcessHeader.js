import React from 'react'
import { Link } from 'react-router-dom/cjs/react-router-dom.min';
import Header from '../../../../components/Header';


const SubProcessHeader = () => {

    const breadCrumbProps = [
        { label: 'PP Management'},
        { 
            label: 'Types',
            items: [
                {
                    key: 1,
                    label: (
                        <Link target="_blank" >
                            PP Management
                        </Link>
                    ),
                },
                {
                    key: 2,
                    label: (
                        <Link target="_blank" >
                            Sub Process
                        </Link>
                    ),
                }
            ],
         },
        { label: 'Sub Process' },
    ];

  return (
    <div>
        <Header
        routes={breadCrumbProps}
        pageTitle={'Sub Process'}
         />
    </div>
  )
}

export default SubProcessHeader