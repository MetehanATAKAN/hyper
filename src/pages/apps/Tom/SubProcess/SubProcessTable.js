import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import TableAccordion from '../../../../components/Tables/TableAccordion';
import Filter from './Filter';
import NewSubProcess from './NewSubProcess';
import Status from '../../../../components/Status';
import Dropdowns from '../../../../components/Dropdowns';
import { ConfirmModal } from '../../../../components/FormElements/InformationModal';

const SubProcessTable = () => {

    const { t } = useTranslation();

    const [tableData, setTableData] = useState([]);

  /**table filter show */
  const [filterShow, setFilterShow] = useState(true);

  /**add button show */
  const [addButtonShow, setAddButtonShow] = useState(false);

  /**action click obj */
  const [obj, setObj] = useState();

  /**delete modal show */
  const [deleteModalShow, setDeleteModalShow] = useState(false);

  /** update modal show */
  const [updateModalShow, setUpdateModalShow] = useState(false);

  /**table loading */
  const [tableLoading, setTableLoading] = useState(false);

  /**is apply */
  const [isApply, setIsApply] = useState(false);

    const statusOptions = [
      {
          id: 9,
          key: 'Edit',
          icon: <i style={{ marginRight: '8px' }} className="fas fa-pen"></i>,
          color: '#6C757D',
          disabled:false
      },
      {
          id: 0,
          key: 'Delete',
          icon: <i style={{ marginRight: '8px' }} className="fas fa-trash"></i>,
          color: '#FA5C7C',
          disabled:true
      },
  ]

  const statusClick = (e,obj,el) => {
    const status = el.key ;
    const data = obj ;
   
   if(status === 'Edit') { // edit
    setUpdateModalShow(true);
    setObj(data);
   }
   else { // delete
    setDeleteModalShow(true);
    setObj(data);
   }
}

  const columns = [
    {
      header: t('ID'),
      accessorKey: 'subProcessId',
      size: '120'
    },
    {
      header: t('Sub Process name'),
      accessorKey: 'subProcessName',
      size: '120'
    },
    {
      header: t('Day'),
      accessorKey: 'day',
      size: '120'
    },
    {
      header: t('Activity Type'),
      accessorKey: 'activityType',
      size: '120'
    },
    {
      header: t('Mix Type'),
      accessorKey: 'mixType',
      size: '120'
    },
    {
      header: t('Process'),
      accessorKey: 'process',
      size: '120'
    },
    {
      header: t('Status'),
      accessorKey: 'status',
      size: '120',
      Cell: ({ cell }) => (
        <Status approveStatus={cell.getValue() === 'Missing' ? 0 : cell.getValue() === 'Redact' ? 1 : 3} />
      ),
    },
    {
      header: '',
      accessorKey: 'action',
      size: '38',
      muiTableBodyCellProps: {
        align: 'center',
      },
      enableColumnActions: true,
      Cell: ({ cell, row }) => {
        return (
          <Dropdowns
            option={statusOptions}
            obj={row.original}
            // option={getStatusOptions(cell.getValue(), row.original)}
            onClick={statusClick}
          />
        )
      }
    }
  ]


  const deleteSubProcess = () => {

  }
      
  return (
    <>

    <TableAccordion
    filter={<Filter setTableData={setTableData} setTableLoading={setTableLoading} isApply={isApply} setIsApply={setIsApply}/>}
    columns={columns}
    data={tableData}
    isBulkButtons={false}
    isCheckBox={false}
    enableExpanding={false}
    filterShow={filterShow}
    setFilterShow={setFilterShow}
    handleNewButton={()=>setAddButtonShow(true)}
    handlApplyBtn={()=>setIsApply(true)}
    isLoading={tableLoading}
    />

    {
        addButtonShow && (
            <NewSubProcess
            show={addButtonShow}
            setShow={setAddButtonShow}
            />
        )
    }

      {
        deleteModalShow && (
          <ConfirmModal
            setShowModal={setDeleteModalShow}
            func={deleteSubProcess}
            message="Are you sure you want to delete this sub process?"
            title="Delete"
          />
        )
      }
    </>
  )
}

export default SubProcessTable