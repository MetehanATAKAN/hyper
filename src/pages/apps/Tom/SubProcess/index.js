import React from 'react'
import SubProcessTable from './SubProcessTable'
import SubProcessHeader from './SubProcessHeader'
import NewSubProcess from './AddNewSubProcess/index'

const SubProcess = () => {
  return (
    <>
    {/* <SubProcessHeader/>
    <SubProcessTable/> */}
    <NewSubProcess/>
    </>
  )
}

export default SubProcess