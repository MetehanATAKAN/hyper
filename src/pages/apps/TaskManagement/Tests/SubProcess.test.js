import * as React from 'react';
import SubProcess from '../SubProcess';
import renderer from 'react-test-renderer';
import { FetchApiGet, FetchApiPost } from '../../../../utils/http.helper';
import { render, fireEvent } from '@testing-library/react';
jest.mock('../../../../utils/http.helper', () => {
    return {
        FetchApiGet: jest.fn(),
        FetchApiPost: jest.fn(),
    };
});

describe('fetch api', () => {
    it('should fetch data', async () => {
        const data = {
            data: [
                [
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                ],
            ],
        };
        FetchApiGet.mockResolvedValueOnce(data);
        const result = await FetchApiGet('services/TaskManagement/SubProcess/GetAllSubProcess', 'GET');
        expect(result).toEqual(data);
    });
});
it('should render correctly', () => {
    const data = {
        data: [
            [
                {
                    id: 1,
                    title: 'test',
                },
                {
                    id: 1,
                    title: 'test',
                },
                {
                    id: 1,
                    title: 'test',
                },
                {
                    id: 1,
                    title: 'test',
                },
            ],
        ],
    };
    FetchApiGet.mockResolvedValue(data);
    const tree = renderer
        .create(<SubProcess updateSubProcess={[]} setOnModal={true} subProcess={data.data} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
describe('input change', () => {
    it('should change input value', () => {
        const data = {
            data: [
                [
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                ],
            ],
        };
        FetchApiGet.mockResolvedValue(data);
        const { getByTestId } = render(<SubProcess updateSubProcess={[]} setOnModal={true} subProcess={data.data} />);
        const input = getByTestId('sub-process-name');
        input.value = 'test';
        expect(input.value).toBe('test');
    });
});
describe('button click', () => {
    it('should call create business process', async () => {
        const data = {
            data: [
                [
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                    {
                        id: 1,
                        title: 'test',
                    },
                ],
            ],
        };
        FetchApiGet.mockResolvedValue(data);
        const { getByTestId } = render(<SubProcess updateSubProcess={[]} setOnModal={true} subProcess={data.data} />);
        const handleChange = jest.fn();
        const input = getByTestId('sub-process-name');
        fireEvent.change(input, { target: { value: 'test' } });
        const button = getByTestId('create-sub-process');
        button.click();
        const postData = {
            subProcessTitle: 'test',
            createdBy: 'string',
            parentProcessId: 0,
        };
        FetchApiPost.mockResolvedValueOnce(postData);
        const result = await FetchApiPost('services/TaskManagement/SubProcess/CreateSubProcess', 'POST', postData);
        expect(result).toEqual(postData);
    });
});
