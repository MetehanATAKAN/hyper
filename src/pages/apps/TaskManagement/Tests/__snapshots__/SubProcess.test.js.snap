// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="parent-process-modal"
>
  <div
    className="parent-process-activity-type"
  >
    <label>
      activity types
    </label>
    <div
      className="react-select css-2b097c-container"
      onKeyDown={[Function]}
    >
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className="react-select__control css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="react-select__value-container react-select__value-container--has-value css-g1d714-ValueContainer"
        >
          <div
            className="react-select__single-value css-1uccc91-singleValue"
          >
            select activity types
          </div>
          <div
            className="css-b8ldur-Input"
          >
            <div
              className="react-select__input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="react-select__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className="react-select__indicator-separator css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="parent-process-job-description"
  >
    <label>
      job description
    </label>
    <div
      className="react-select react-select--is-disabled css-14jk2my-container"
      onKeyDown={[Function]}
    >
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className="react-select__control react-select__control--is-disabled css-1fhf3k1-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="react-select__value-container react-select__value-container--has-value css-g1d714-ValueContainer"
        >
          <div
            className="react-select__single-value react-select__single-value--is-disabled css-107lb6w-singleValue"
          >
            select job description
          </div>
          <div
            className="css-1bwsx9e-Input"
          >
            <div
              className="react-select__input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={true}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="react-select__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className="react-select__indicator-separator css-109onse-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="parent-process-business-process"
  >
    <label>
      business process
    </label>
    <div
      className="react-select react-select--is-disabled css-14jk2my-container"
      onKeyDown={[Function]}
    >
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className="react-select__control react-select__control--is-disabled css-1fhf3k1-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="react-select__value-container react-select__value-container--has-value css-g1d714-ValueContainer"
        >
          <div
            className="react-select__single-value react-select__single-value--is-disabled css-107lb6w-singleValue"
          >
            select business process
          </div>
          <div
            className="css-1bwsx9e-Input"
          >
            <div
              className="react-select__input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={true}
                id="react-select-4-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="react-select__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className="react-select__indicator-separator css-109onse-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="parent-process-business-process"
  >
    <label>
      Parent process
    </label>
    <div
      className="react-select react-select--is-disabled css-14jk2my-container"
      onKeyDown={[Function]}
    >
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className="react-select__control react-select__control--is-disabled css-1fhf3k1-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="react-select__value-container react-select__value-container--has-value css-g1d714-ValueContainer"
        >
          <div
            className="react-select__single-value react-select__single-value--is-disabled css-107lb6w-singleValue"
          >
            select parent process
          </div>
          <div
            className="css-1bwsx9e-Input"
          >
            <div
              className="react-select__input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={true}
                id="react-select-5-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="react-select__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className="react-select__indicator-separator css-109onse-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="parent-process-add-sub"
  >
    <label>
      add sub process
    </label>
    <input
      className="form-control"
      data-testid="sub-process-name"
      disabled={true}
      onChange={[Function]}
      placeholder="add sub process"
      type="text"
      value=""
    />
  </div>
  <div
    className="task-management-footer-btn"
  >
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      cancel
    </button>
    <button
      className="btn btn-primary"
      data-testid="create-sub-process"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      add
    </button>
  </div>
</div>
`;
